src/history/hs.c

```diff
@@ -1880,27 +1896,26 @@ __wt_history_store_verify(WT_SESSION_IMPL *session)
          * The cursor is positioned either from above or left over from the internal call on the
          * first key of a new btree id.
          */
-        WT_ERR(cursor->get_key(cursor, &btree_id, &hs_key, &hs_start_ts, &hs_counter));
+        WT_ERR(hs_cursor->get_key(hs_cursor, &btree_id, &key, &hs_start_ts, &hs_counter));
         if ((ret = __wt_metadata_btree_id_to_uri(session, btree_id, &uri_data)) != 0) {
             F_SET(S2C(session), WT_CONN_DATA_CORRUPTION);
             WT_ERR_PANIC(session, WT_PANIC,
-              "Unable to find btree id %" PRIu32
-              " in the metadata file for the associated history store key %s",
-              btree_id, __wt_buf_set_printable(session, hs_key.data, hs_key.size, buf));
+              "Unable to find btree id %" PRIu32 " in the metadata file for the associated key %s",
+              btree_id, __wt_buf_set_printable(session, key.data, key.size, buf));
         }
-        WT_ERR(__wt_open_cursor(session, uri_data, NULL, NULL, &data_cursor));
-        F_SET(data_cursor, WT_CURSOR_RAW_OK);
-        ret = __verify_history_store_id(session, (WT_CURSOR_BTREE *)data_cursor, btree_id);
+        WT_ERR(__wt_open_cursor(session, uri_data, NULL, NULL, &ds_cursor));
```

Comment:
@agorrod This call returns ebusy because the btree has WT_BTREE_VERIFY flag set which is one of the flags classified as WT_BTREE_SPECIAL_FLAGS.
```
    /*
     * If this session already has exclusive access to the handle, there is no point trying to lock
     * it again.
     *
     * This should only happen if a checkpoint handle is locked multiple times during a checkpoint
     * operation, or the handle is already open without any special flags. In particular, it must
     * fail if attempting to checkpoint a handle opened for a bulk load, even in the same session.
     */
    if (dhandle->excl_session == session) {
        if (!LF_ISSET(WT_DHANDLE_LOCK_ONLY) &&
          (!F_ISSET(dhandle, WT_DHANDLE_OPEN) ||
            (btree != NULL && F_ISSET(btree, WT_BTREE_SPECIAL_FLAGS))))
                return (__wt_set_return(session, EBUSY));
            
        ++dhandle->excl_ref;
        return (0);
    }
```